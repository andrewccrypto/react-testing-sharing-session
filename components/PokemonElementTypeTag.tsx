import { ReactElement } from 'react';
import {
  PokemonElement,
  PokemonElementBug,
  PokemonElementDark,
  PokemonElementDragon,
  PokemonElementElectric,
  PokemonElementFairy,
  PokemonElementFighting,
  PokemonElementFire,
  PokemonElementFlying,
  PokemonElementGhost,
  PokemonElementGrass,
  PokemonElementGround,
  PokemonElementIce,
  PokemonElementNormal,
  PokemonElementPoison,
  PokemonElementPsychic,
  PokemonElementRock,
  PokemonElementSteel,
  PokemonElementWater,
} from '@types';

interface PokemonElementTypeTagProps {
  element: PokemonElement;
}

function PokemonElementTypeTag({
  element,
}: PokemonElementTypeTagProps): ReactElement {
  function getBackgroundColor() {
    switch (element) {
      case PokemonElementBug:
        return '#A8B820';
      case PokemonElementDark:
        return '#705848';
      case PokemonElementDragon:
        return '#7038F8';
      case PokemonElementElectric:
        return '#F8D030';
      case PokemonElementFairy:
        return '#EE99AC';
      case PokemonElementFighting:
        return '#C03028';
      case PokemonElementFire:
        return '#F08030';
      case PokemonElementFlying:
        return '#A890F0';
      case PokemonElementGrass:
        return '#78C850';
      case PokemonElementGhost:
        return '#705898';
      case PokemonElementGround:
        return '#E0C068';
      case PokemonElementIce:
        return '#98D8D8';
      case PokemonElementNormal:
        return '#A8A878';
      case PokemonElementPoison:
        return '#A040A0';
      case PokemonElementPsychic:
        return '#F85888';
      case PokemonElementRock:
        return '#B8A038';
      case PokemonElementSteel:
        return '#B8B8D0';
      case PokemonElementWater:
        return '#6890F0';
    }
  }

  return (
    <div
      style={{
        backgroundColor: getBackgroundColor(),
        borderRadius: '5px',
        color: '#fff',
        display: 'inline-block',
        padding: '5px',
      }}
    >
      {element}
    </div>
  );
}

export default PokemonElementTypeTag;
