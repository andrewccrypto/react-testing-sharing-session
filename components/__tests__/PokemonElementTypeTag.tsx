import { render } from "@testing-library/react";
import PokemonElementTypeTag from "@components/PokemonElementTypeTag";
import { PokemonElementBug, PokemonElementGrass } from "@types";

describe("PokemonElementTypeTag", () => {
  it("Should handle Bug element type", () => {
    const { container } = render(
      <PokemonElementTypeTag element={PokemonElementBug} />
    );

    expect(container).toMatchSnapshot();
  });

  it("Should handle Grass element type", () => {
    const { container } = render(
      <PokemonElementTypeTag element={PokemonElementGrass} />
    );

    expect(container).toMatchSnapshot();
  });
});
