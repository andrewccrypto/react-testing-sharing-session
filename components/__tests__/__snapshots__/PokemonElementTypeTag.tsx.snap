// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonElementTypeTag Should handle Grass element type 1`] = `
<div>
  <div
    style="background-color: rgb(120, 200, 80); border-radius: 5px; color: rgb(255, 255, 255); display: inline-block; padding: 5px;"
  >
    grass
  </div>
</div>
`;
